include/native/parallel_execution_native.h

```diff
@@ -32,105 +32,236 @@
 
 namespace grppi {
 
-//extern bool initialised;
-//extern thread_pool pool;
-/** @brief Set the execution mode to parallel with posix thread implementation 
- */
-struct parallel_execution_native {
-  public: 
-  thread_pool pool;
-  constexpr static int default_queue_size = 100;
-  int queue_size = default_queue_size;
-  queue_mode lockfree = queue_mode::blocking;
+/**
+\brief Thread index table to provide portable natural thread indices.
 
-  void set_queue_size(int new_size){
-     queue_size = new_size;
-  }
+A thread table provides a simple way to offer thread indices (starting from 0).
 
+When a thread registers itself in the registry. To get an integer index, users
```

Comment:
Rewritten as:

A thread table provides a simple way to offer thread indices (starting from 0).

When a thread registers itself in the registry, its id is added to the vector 
of identifiers. When a thread deregisters itself from the registry its entry
is modified to contain the empty thread id.

To get an integer index, users may call `current_index`, which provides the order
number of the calling thread in the registry.